src/storage/store/engine.rs

```diff
@@ -167,7 +171,15 @@ impl ShardEngine {
         );
 
         if hashes_match {
-            self.trie.commit().unwrap();
+            {
+                let mut write_guard = self.db.db.write().unwrap();
+                if let Some(db) = write_guard.as_mut() {
+                    let mut txn = db.transaction();
+                    self.trie.commit_to_txn(&mut txn);
+                    txn.commit().unwrap();
```

Comment:
I think the control should be the other way around, we provide it the RocksDBTransaction batch and we commit it outside the trie. The stores also operate on the RocksDBTransaction batch